google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionClient.java

```diff
@@ -224,6 +225,39 @@ SessionImpl createSession() {
     }
   }
 
+  /**
+   * Create a multiplexed session and returns it to the given {@link SessionConsumer}. A multiplexed
+   * session is not affiliated with any GRPC channel. The given {@link SessionConsumer} is
+   * guaranteed to eventually get exactly 1 multiplexed session unless an error occurs. In case of
+   * an error on the gRPC calls, the consumer will receive one or more {@link
+   * SessionConsumer#onSessionCreateFailure(Throwable, int)} calls with the error.
+   *
+   * @param consumer The {@link SessionConsumer} to use for callbacks when sessions are available.
+   */
+  void createMultiplexedSession(SessionConsumer consumer) {
```

Comment:
Yes, I initially started by keeping the method behavior to be similar to existing `createSession` method which is also synchronous but it does not take any consumer. Thinking about how this method will be used a) in background thread where it will invoked in background but we would block for a session to get created because otherwise we cannot remove a stale session and b) In `BatchClient` where also a sync behavior is needed.

Reply:
Let's leave it like this for now, but it might be that this should be changed at a later moment. Taking a `Consumer` as an input argument for a sync method is not really necessary. Instead, the method should just return the `SessionImpl`.
But considering that we are likely to use this method in the initialization in the Session pool, then I think that it will need to be changed to be async, so it can be called in the same way as `batchCreateSessions` can be called.